### PR TITLE
fix: ongoing call foreground service crash

### PR DESCRIPTION
**Pull request by alexandreferris**
----
#### PR Submission Checklist for internal contributors

- The **PR Title**
  - [X] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
  - [X] contains a reference JIRA issue number like `SQPIT-764`
  - [X] answers the question: _If merged, this PR will: ..._ ³

- The **PR Description**
  - [X] is free of optional paragraphs and you have filled the relevant parts to the best of your ability
----

# What's new in this PR?

[AR-2475 - Calling: minimizing the app stopped the call. Trying to reconnect crashed the app](https://wearezeta.atlassian.net/browse/AR-2475)

### Issues

App is crashing with `ForegroundServiceStartNotAllowedException`

### Causes (Optional)

If user has Android 12 device running the app, when in an ongoing call and minimizing the app (putting app into background) the app would crash due to the ongoingCall service notification not meeting the required specifications to run a foreground service while in the background.

TLDR: Service Notification was a phone call but was not categorized as one.

### Solutions

Add `PHONE_CALL` category to both notification and service specification in the Manifest.

----
##### References
1. https://developer.android.com/guide/components/foreground-services#background-start-restriction-exemptions
2. https://developer.android.com/guide/components/foreground-services#notification-immediate
3. https://developer.android.com/guide/topics/manifest/service-element#foregroundservicetype
4. https://developer.android.com/reference/android/app/Notification#category
5. Last but not least from SO: https://stackoverflow.com/a/70666991

### CODE REVIEW

**Comment by AndroidBob** [comment]
Build [1](https://10.10.124.134/job/wire-android-reloaded/job/PR-1005/1/) **failed**.

**Comment by AndroidBob** [comment]
Build [2](https://10.10.124.134/job/wire-android-reloaded/job/PR-1005/2/) **failed**.

**Review by yamilmedina — APPROVED** [review]
Thanks for adding the references pointing to documentation. Great job as usual ! 👏

**Comment by AndroidBob** [comment]
Build [3](https://10.10.124.134/job/wire-android-reloaded/job/PR-1005/3/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-vv0.0.1(4471038)(4471038)-dev-release.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/dev/release/com.wire.android-vv0.0.1(4471038)(4471038)-dev-release.apk)

**Comment by AndroidBob** [comment]
Build [4](https://10.10.124.134/job/wire-android-reloaded/job/PR-1005/4/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-vv0.0.1(4472040)(4472040)-dev-release.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/dev/release/com.wire.android-vv0.0.1(4472040)(4472040)-dev-release.apk)
